app/assets/javascripts/test/puppeteer/screenshot_helpers.js

```diff
@@ -0,0 +1,69 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+const pixelmatch = require("pixelmatch");
+const { PNG } = require("pngjs");
+const fs = require("fs");
+
+function openScreenshot(path, name) {
+  return new Promise(resolve => {
+    fs
+      .createReadStream(`${path}/${name}.png`)
+      .on("error", error => {
+        if (error.code === "ENOENT") {
+          resolve(null);
+        } else {
+          throw error;
+        }
+      })
+      .pipe(new PNG())
+      .on("parsed", function() {
+        resolve(this);
+      });
+  });
+}
+
+function saveScreenshot(png, path, name) {
+  return new Promise(resolve => {
+    png
+      .pack()
+      .pipe(fs.createWriteStream(`${path}/${name}.png`))
+      .on("finish", () => resolve());
+  });
+}
+
+function bufferToPng(buffer, width, height) {
+  return new Promise(resolve => {
+    const png = new PNG({ width, height });
+    png.parse(buffer, () => resolve(png));
+  });
+}
+
+async function compareScreenshot(screenshotBuffer, width, height, path, name) {
+  const newScreenshot = await bufferToPng(screenshotBuffer, width, height);
```

Comment:
Can you parallelize the awaits in 41 + 42 (and 61 + 62) via `Promise.all`? Should make it a bit faster, since it's apparently cpu bound in the moment :)